File: frontend/lib/feedback.ts
import { useMutation, useQuery } from "@tanstack/react-query";

import { apiFetch } from "./api";

export type FeedbackCategory = "bug" | "suggestion";

export interface FeedbackEvent {
  kind: "status" | "reply";
  status: string | null;
  body: string | null;
  created_at: string;
}

export interface FeedbackThread {
  id: string;
  category: FeedbackCategory | string;
  content: string;
  image_url: string | null;
  status: string;
  created_at: string;
  events: FeedbackEvent[];
}

export interface FeedbackPayload {
  category: FeedbackCategory;
  content: string;
  image?: string | null; // base64 data URL
  page_url?: string | null;
  contact?: string | null;
}

const submitFeedback = (payload: FeedbackPayload) =>
  apiFetch<{ id: string }>("/api/feedback", {
    method: "POST",
    body: JSON.stringify(payload),
  });

/** 提交反馈。成功后调用方自行关闭弹窗 / 提示。 */
export function useSubmitFeedback() {
  return useMutation({ mutationFn: submitFeedback });
}

/** 反馈线程（本体 + 时间线）。用于通知详情里展示「解决记录」全貌。 */
export function useFeedbackThread(feedbackId: string | null) {
  return useQuery({
    queryKey: ["feedback-thread", feedbackId],
    queryFn: () => apiFetch<FeedbackThread>(`/api/feedback/${feedbackId}`),
    enabled: !!feedbackId,
    staleTime: 10_000,
  });
}

File: backend/schemas/feedback.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


# ---------- 用户端 ----------

class FeedbackCreateIn(BaseModel):
    category: str  # bug / suggestion
    content: str = Field(min_length=1, max_length=4000)
    image: str | None = None  # base64 data URL（可选截图）
    page_url: str | None = Field(default=None, max_length=500)
    contact: str | None = Field(default=None, max_length=200)


# ---------- admin ----------

class FeedbackAdminOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str | None = None
    category: str
    content: str
    image_url: str | None = None
    page_url: str | None = None
    contact: str | None = None
    user_agent: str | None = None
    status: str
    admin_note: str | None = None
    reply: str | None = None
    created_at: datetime
    updated_at: datetime


class FeedbackUpdateIn(BaseModel):
    status: str | None = None  # new / triaged / resolved
    admin_note: str | None = None  # 内部备注
    reply: str | None = None  # 回复用户（对外，进通知）


# ---------- 用户线程视角 ----------

class FeedbackEventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    kind: str  # status / reply
    status: str | None = None
    body: str | None = None
    created_at: datetime


class FeedbackThreadOut(BaseModel):
    id: str
    category: str
    content: str
    image_url: str | None = None
    status: str
    created_at: datetime
    events: list[FeedbackEventOut]
